src/sentry/models/organizationmapping.py

```diff
@@ -34,3 +34,22 @@ class Meta:
         db_table = "sentry_organizationmapping"
 
     __repr__ = sane_repr("organization_id", "slug", "region_name", "verified")
+
+    @classmethod
+    def find_region_name_by_org_id(cls, id: int) -> str | None:
+        mapping: OrganizationMapping = OrganizationMapping.objects.filter(
```

Comment:
This demonstrates how we might go from slug or org ids -> region name, which can then be mapped by our region configs to region objects, and then region endpoints.

Reply:
We are probably doing this quite a bit, i believe this is a candidate for going through the caching functionality of the models. (`objects.get_from_cache`). Same with the query below.

Reply:
Honestly, it's likely this can be entirely cached in memory so I agree we could use a caching strategy.  One open question is how we'll handle cache validation though -- in the org `a` renames to `c` and org `b` renames to `a`.

Reply:
This is doubly tricky because if we contact the wrong region, we won't necessarily have feedback that we did -- we just report the wrong results.  So whatever caching we do use has to be precise.  It's acceptable to occasionally report wrong results.

Reply:
Our base model managers can bake cache refresh/purges in for models that have unique keys like id and slugs. We should be able to leverage that here as well.

Reply:
We can reconsider better caches later but the basic `get_from_cache` functionality is all quite consistently used throughout the system today, I would leverage this for the first iteration.